### PR TITLE
Define graphics and print abstractions in WinPrint.Core

### PR DESCRIPTION
**Pull request by tig**
## Summary
- add IGraphicsContext and graphics primitives in WinPrint.Core.Abstractions
- add IPrintService abstraction for printer discovery, page setup, and job lifecycle
- add SystemDrawingGraphicsContext adapter so existing System.Drawing rendering remains functional
- refactor ContentTypeEngineBase.PaintPage to use IGraphicsContext
- update TextCte, TextMateCte, and header/footer rendering pipeline (SheetViewModel, HeaderFooterViewModel) to use the abstraction
- update stubbed AnsiCte and HtmlCte paint signatures to IGraphicsContext

## Validation
- MSBuild.exe tests/WinPrint.Core.UnitTests/WinPrint.Core.UnitTests.csproj /t:Build /p:Configuration=Debug /p:Platform=x64 /v:q /nologo
- dotnet test tests/WinPrint.Core.UnitTests/WinPrint.Core.UnitTests.csproj --no-build -c Debug -p:Platform=x64
- result: 57 total, 55 passed, 2 skipped, 0 failed

### CODE REVIEW

File: src/WinPrint.Core/Abstractions/SystemDrawingGraphicsContext.cs [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        public static GraphicsRectF FromRectangleF(RectangleF rect) {
+            return new GraphicsRectF(rect.X, rect.Y, rect.Width, rect.Height);
+        }
+
+        public static RectangleF ToRectangleF(GraphicsRectF rect) {
+            return new RectangleF(rect.X, rect.Y, rect.Width, rect.Height);
+        }
+
+        private static Rectangle ToRectangle(GraphicsRectF rect) {
+            return new Rectangle((int)Math.Round(rect.X), (int)Math.Round(rect.Y), (int)Math.Round(rect.Width), (int)Math.Round(rect.Height));
```
[the remark this answers is not in the document]
Comment (tig) [reply src/WinPrint.Core/Abstractions/SystemDrawingGraphicsContext.cs]: Good catch. I applied this by keeping clip bounds as floats: SetClip now uses RectangleF directly, and ExcludeClip now uses a Region created from RectangleF so sub-pixel clip edges are preserved.

**Comment by tig** [comment]
## Code Review: PR #58 "Define graphics and print abstractions in WinPrint.Core"

**Summary**: This is a very large PR (114 files, ~12k insertions) that delivers the graphics abstraction foundation for the MAUI port while also introducing a comprehensive code quality infrastructure overhaul (editorconfig, Directory.Build.props with TreatWarningsAsErrors + custom analyzers, new CI, global reformatting, type splitting, submodule→NuGet, nullable fixes, etc.).

**What I like**:
- The `IGraphicsContext` + primitive types + `SystemDrawingGraphicsContext` adapter is a clean abstraction layer for the GDI+ painting path used by CTEs, SheetViewModel, headers/footers, and PrintPreview. This is exactly the right first step toward platform-agnostic rendering.
- IPrintService/IPrintJob declarations show forward thinking for printer abstraction.
- Code quality changes (gui-cs-inspired style, one-type-per-file + no-nested-types via Roslyn analyzer WPA0001/WPA0002, CI style gates with `jb cleanupcode` + `dotnet format`, LF normalization) will pay dividends long-term and align with other Kindel projects.
- Replacing the Terminal.Gui.Cli git submodule with the `Terminal.Gui.Cli` NuGet package dramatically improves buildability and CI reliability.
- Behavior appears preserved: tests still pass at the same rate as develop (modulo the pre-existing Pygments failure).
- Lots of cleanup: split multi-type files, promoted types, fixed nullable/analyzer warnings.

**Issues / Required fixes**:

1. **Critical: Analyzer fails to load when building with `-p:Platform=x64` (or `/p:Platform=x64` in some cases)**  
   Reproducible with `dotnet build ... -p:Platform=x64`:
   ```
   CSC : error CS8034: Unable to load Analyzer assembly ...WinPrint.Analyzers\bin\x64\Debug\... : architecture is not compatible with the current process architecture.
   ```
   The analyzer csproj declares `<Platforms>AnyCPU</Platforms>` but the Directory.Build.props ProjectReference + incoming Platform=x64 causes MSBuild to emit the netstandard assembly under `bin\x64\...`, and the 64-bit dotnet host refuses to load it.  
   Your CI validation used `MSBuild.exe ... /p:Platform=x64` which may have hidden this. `dotnet build` (common for devs and some CI) breaks.  
   **Fix**: Ensure the analyzer is always built/loaded as architecture-neutral. Options: force `<PlatformTarget>AnyCPU</PlatformTarget>` in the analyzer csproj (or via props for that project), use `SetTargetPlatform` or condition the reference, or build the analyzer project explicitly without platform in the props. Test both `dotnet build -p:Platform=x64` and plain `dotnet build`.

2. **PR scope is too broad - feature + massive refactor mixed**  
   The abstractions + CTE/SheetViewModel painting refactor is ~20-30 files. The rest is infrastructure + global format + splits. This makes meaningful review difficult (many formatting-only changes obscure the semantic diffs in SheetViewModel.cs, ContentTypeEngineBase.cs, TextMateCte.cs, etc.).  
   **Recommendation**: Split future work. Land quality infra (editorconfig/Directory.Build.props/analyzers/CI/format) in one PR, then the functional graphics changes in a follow-up (or stacked). At minimum, call out the non-functional commits clearly in the PR description.

3. **Abstraction is incomplete / leaky**  
   - `TextMateCte.PaintPage` (and likely others) does:
     ```cs
     if (graphicsContext is not SystemDrawingGraphicsContext context)
         throw new NotSupportedException("TextMateCte currently requires a System.Drawing graphics context.");
     Graphics g = context.Graphics;
     ```
     The I* interfaces are not yet rich enough to replace direct Graphics usage for syntax-highlighted rendering.  
   - `IPrintService` / `IPrintJob` / `PrintDialogOptions` etc. are declared but **never implemented or used**. `Print.cs`, WinForms, and CLI still talk directly to `System.Drawing.Printing.PrintDocument`, `PrinterSettings`, `PrintDialog`, etc.  
   **Action**: Either (a) implement a `SystemDrawingPrintService` adapter in this PR, or (b) document explicitly in the PR description and a TODO/ROADMAP that this is partial and printer/print-job abstraction is deferred. For the MAUI goal, a real cross-platform print path will also need this.

4. **No tests for the new abstraction layer**  
   Zero tests were added for `IGraphicsContext`, the primitive structs, the adapter, or the new `DelayedEvent` helper. The existing CTE/ViewModel tests exercise the path indirectly (good), but dedicated unit tests (especially for the adapter round-tripping and any new edge cases in clip/transform/measure) would increase confidence in the refactor. Add at least smoke + a few behavioral tests.

5. **Pygments test still fails on machines without Python** (same as develop)  
   `PygmentsConverterServiceTests.ConvertAsyncTest` throws `InvalidOperationException` instead of being skipped like the AnsiCte render tests. The PR validation claimed "0 failed". Make the test robust (e.g. `[SkippableFact]` or conditional skip) so clean `dotnet test` passes in minimal envs.

6. **Graphics primitive structs are very minimal**  
   `GraphicsPointF`, `GraphicsRectF`, `GraphicsSizeF`, `GraphicsColor`, etc. are plain mutable structs with only properties/ctor. No `IEquatable<T>`, `==`/`Equals`, `GetHashCode`, `Deconstruct`, `ToString`, or record semantics. For types that will be used in calculations, collections, and comparisons during layout, this is a footgun. Consider making them `readonly record struct` (or at least adding the standard value-type boilerplate).

7. **Style / tooling nits**
   - The new "space before parens" rule (`Method (arg)`) + file-scoped ns + brace rules are applied consistently via format + jb cleanup. This is a big culture shift; document it in CONTRIBUTING.md or a style guide.
   - `Directory.Build.props` sets `ImplicitUsings=disable` globally. PR #57's Maui csproj sets `enable`. Verify precedence and that the Maui project still builds cleanly under the new props (precedence, usings, analyzers) and that `dotnet format` / jb cleanup don't fight the Maui template style.
   - `.config/dotnet-tools.json` pins `jb` (JetBrains.ReSharper.GlobalTools?). Make sure the version is stable and the tool is available in the CI image.
   - The analyzer exempts `*.Designer.cs` etc. - good, but also consider exempting generated XAML code in the future Maui project.

8. **Other observations**
   - Great that you kept the `SystemDrawingGraphicsContext` as an internal adapter rather than forcing all call sites to change immediately.
   - The `WrappedLine` / `TextMateWrapped*` new types from the refactor look like good extraction.
   - File watcher changes (`FileSystemSafeWatcher`, `DelayedEvent`) are non-trivial; the disclaimer in DelayedEvent ("NOT been tested") is concerning given it's now in the hot path. Add tests or remove the disclaimer after validation.
   - Telemetry, logging, and Settings paths were touched heavily in the "fix nullable" and format passes - spot-check a couple for behavioral changes.

**Testing performed (on this machine)**:
- develop: 54 pass, 2 skip (Ansi), 1 fail (Pygments test)
- PR58 (no Platform=x64): clean build (0 warnings, thanks to TreatAsErrors + fixes), same test results
- With `-p:Platform=x64`: analyzer load failure (see #1)

**Recommendation**: Fix #1 (analyzer) and #5 (test skip) as must-haves. Strongly consider splitting the PR or at least a very clear "this commit is pure refactor" narrative + review focus on the 3-4 files with real logic changes (ContentTypeEngineBase.cs, SheetViewModel.cs, the CTEs, SystemDrawingGraphicsContext.cs, Print.cs changes).

Overall direction is excellent for the MAUI port (#54). Once the build hygiene and scope issues are addressed, this is a big step forward.

(Reviewed against develop @ 06fa2a0 + worktree of PR branch. Full context in local review.)
